frontend/document/credocument.lua

```diff
@@ -26,6 +26,7 @@ local CreDocument = Document:new{
     fallback_font = G_reader_settings:readSetting("fallback_font") or "Noto Sans CJK SC",
     default_css = "./data/cr3.css",
     options = CreOptions,
+    color = G_reader_settings:isTrue("color_rendering"),
```

Comment:
I think we should do something slightly different here. More properly a Device method like `hasColor` (there is also such a thing as color E Ink). That could be combined with the setting. Brainstorm:

```lua
G_reader_settings:has("color_rendering") and G_reader_settings:isTrue("color_rendering") or Device:hasColor()
```

That way you can enable it on gray scale (why? no idea…) or disable it on Android (why? um, emulating the E Ink experience? :-P)

As an aside, I think we should consider integrating some Kobo-like device specific code on Android. E.g., Onyx stuff hasColor return false. But also some of the stuff people stick in patch.lua.

Reply:
PS Actually the `color_rendering` setting could just be part of the Device method.
```lua
function Device:hasColor()
    return G_reader_settings:has("color_rendering") and G_reader_settings:isTrue("color_rendering") or false -- true on Android
end

Reply:
(I didn't go further into that, I was just allowing us testing this with an hidden setting).
But agreed with what you said.
I think we need just allow people on color devices to get back their greyscale experience :)
And there's also may be the memory limitation (so there may be old colored android that are better with the small greyscale blitbuffer), althought I dunno, that may be negligable compared to all other memory stuff.
Or the following ?
```lua
function Device:wantsColor()
    if Device.hasColor and G_reader_settings:isFalse("color_rendering") return false
    if not Device.hasColor and G_reader_settings:isTrue("color_rendering") return true
    return Device.hasColor
end
```

Reply:
Anyway, if you don't have a better name than `color_rendering` for the setting, and if you, like me, feel this PR is a no-op as long as this setting is not set, we could merge in base, + bump base here and merge, so we can test it on the next nightly apk.
That device setting & menu could be part of another PR (and I wouldn't mind if you'd care about that :)

Reply:
Perhaps, though to me that's more overcomplicated and confusing. If you're unhappy with what I wrote initially I'd suggest something more like

```lua

if G_reader_settings:has("color_rendering") then return G_reader_settings:isTrue("color_rendering") end
return Device.screen_has_color
```

In generic Device
```lua
Device.screen_has_color = false


```

In specific init
```lua
init()
self.screen_has_color = true -- if necessary
```

Reply:
OK, but for now, I'd like to replace all the `G_reader_settings:isTrue("color_rendering")` with
SomeStuff:someMethod(), that would for now just `return G_reader_settings:isTrue("color_rendering")`, and later the logic you suggested about device.screen_has_color and this setting.
Device:wantsColor() ? something else than Device ?

Reply:
I don't really care for wantsColor.

hasColor(Screen) would fit right in with hasKeyboard, hasFrontlight, etc. and definitely has my preference

isColorEnabled (or something along those lines) would fit in with isTouchDevice

But yes, regardless of the name I'd say `Device:hasColorScreen() return isTrue("color_rendering")` should do the trick for now.

Reply:
Or should I just as you initially suggested just use Device:hasColor() even if some decision based on a menu/setting will be made there?

Reply:
Doesn't really seem like much of an issue to me. But maybe we should just skip the function and do it like this:

```lua
    local is_eink = G_reader_settings:readSetting("eink")
    self.screen.eink = (is_eink == nil) or is_eink
```

Reply:
Would that do (explicitely separating device capabilities and overiding setting) ?
`color = Device.hasColorScreen() and G_reader_settings:isFalse("color_rendering")` in all these PR files.
And Device.hasColorScreen = no in generic/device.lua , and = yes in android + sdl/device.lua?

Reply:
Damned, there is a problem with comment visibilité on github today ... I see your comments older than mine minutes after I post mine...
Yeah, the is_eink way looks best.

Reply:
That's the wrong way around. It should be my original suggestion in that case, but I'd prefer not littering the individual stuff with that for clarity and duplication. 

How about something like this?

```lua
self.screen.color = false -- probably shouldn't be used by any code anywhere though?
self.screen.isColorEnabled = function() G_reader_settings:readSetting("color_rendering") or self.color end
```

Reply:
The heck? I also didn't see your comment from 21 minutes ago until just yet.

Anyway, what do you think of my last comment from 21:37?

Reply:
I like your 21:37 suggestion, I'm doing that.